src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -590,6 +590,79 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context, bool l
             return this.CurrentLocation;
         }
 
+        /// <summary>
+        /// Changes the current working directory to the path specified.
+        /// It is only for FileSystem provider paths and optimized to be fast.
+        /// </summary>
+        /// <param name="path">
+        /// The file system path of the new current working directory.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// If <paramref name="path"/> is null.
+        /// </exception>
+        /// <exception cref="DriveNotFoundException">
+        /// If <paramref name="path"/> refers to a drive that does not exist.
+        /// </exception>
+        internal void SetLocationFileSystemFast(string path)
+        {
+            if (path == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(path));
+            }
+
+            PathInfo current = null;
+            if (PublicSessionState.InvokeCommand.LocationChangedAction != null)
+            {
+                // It is under a condition to avoid the unneeded allocation - CurrentLocation always allocates.
```

Comment:
I don't understand your comment.  The two parts of the sentence seem to conflict.  How about this:
```suggestion
                // Allocate a new location using the CurrentLocation property
```

Reply:
I want protect the condition from removing because CurrentLocation always allocates and we should avoid the allocation if we don't use CurrentLocation in code below. Your proposal doesn't reflect this.
What about "CurrentLocation always allocates and we should avoid the allocation if we don't use CurrentLocation later."?

Reply:
I can understand that much better.  Please use that.